compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/SkiaTextInputService.skiko.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.platform
+
+import androidx.compose.ui.text.input.CommitTextCommand
+import androidx.compose.ui.text.input.EditCommand
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.ImeOptions
+import androidx.compose.ui.text.input.PlatformTextInputService
+import androidx.compose.ui.text.input.TextFieldValue
+
+class SkiaTextInputService(private val inputObserver: InputObserver) : PlatformTextInputService {
+
+    var listener: ((String) -> Unit)? = null
+
+    override fun startInput(
+        value: TextFieldValue,
+        imeOptions: ImeOptions,
+        onEditCommand: (List<EditCommand>) -> Unit,
+        onImeActionPerformed: (ImeAction) -> Unit
+    ) {
+        listener = { text: String ->
```

Comment:
If the `startInput` can be called only once before the next `stopInput`, then we don't need `textInputListeners = mutableSetOf<(String) -> Unit>` (and so `InputObserver`).

We probably can just keep the field here, and make a method `SkiaTextInputService.sendInputEvent`, which can be called directly, without inderection via `InputObserver`:
```
private val inputService = SkiaTextInputService()
private val platform = object : Platform by Platform.Empty {
    override val textInputService = inputService
}

inner class ComponentImpl : SkikoView, Platform by platform {
  override fun onInputEvent(event: SkikoInputEvent) {
     inputService.sendInputEvent(event)
  }
```

Reply:
Good Idea!
done